### PR TITLE
Settings page

### LINKED ISSUE #53: [Settings page] Make the settings form work
### Features
* Move `app/templates/settings.hbs` template content into a `{{settings-form}}` component
* Make sure all fields are bound to the `user` model
* Disable the button unless the model is dirty
* Make the submit button work
* Navigating away from the page should revert model change (see https://github.com/patocallaghan/realworld-starter-kit/blob/4611cda2c1b9f391f976b111e2e23a833d3dd00c/app/components/settings-form.js#L8-L13)

image

---
#### Update user
PUT /api/user

Example request body:
```
{
  "user":{
    "email": "jake@jake.jake",
    "bio": "I like to skateboard",
    "image": "https://i.stack.imgur.com/xHWG8.jpg"
  }
}
```
Authentication required, returns the User

Accepted fields: email, username, password, image, bio

This should be handled by `user.save()`.

---

### Tests
* Integration - Submit button is disabled until content is changed
* Integration - Updating content sends the correct http requests

Octane example: https://github.com/patocallaghan/realworld-starter-kit/blob/4611cda2c1b9f391f976b111e2e23a833d3dd00c/app/templates/components/settings-form.hbs

### LINKED ISSUE #54: [Settings page] Log out & redirect to homepage
### Features
* Make the log out button work, i.e. `this.session.logOut()`
* Redirect to the homepage afterwards on a successful response

### Tests
* Acceptance - log out and redirects to the homepage

### LINKED ISSUE #55: [Settings page] Errors
### Features
* display errors on the page

image

Want to write tests?
* Integration: submit invalid data and errors are shown

Example from Octane app: https://github.com/patocallaghan/realworld-starter-kit/blob/4611cda2c1b9f391f976b111e2e23a833d3dd00c/app/templates/components/settings-form.hbs#L3-L9

### PR DESCRIPTION
**Pull request by GCheung55**
Fixes #53 - [Settings page] Make the settings form work
* [x] Move app/templates/settings.hbs template content into a {{settings-form}} component
* [x] Make sure all fields are bound to the user model
* [x] Disable the button unless the model is dirty
* [x] Make the submit button work
* [x] Navigating away from the page should revert model change

Fixes #54 - [Settings page] Log out & redirect to homepage
* [x] Make the log out button work, i.e. `this.session.logOut()`
* [x] Redirect to the homepage afterwards on a successful response

Fixes #55 [Settings page] Errors
* [x] display errors on the page

### CODE REVIEW

**Comment by GCheung55** [comment]
There’s a failing test which I will address. 

There’s some code shared between this PT and #85. I can refactor after either is merged or after both are merged. 

My preference would be after both are merged so we don’t block the project from being feature complete.

**Comment by Alonski** [comment]
After I merged in #85 there is a conflict now :)

File: app/components/settings-form.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export default Component.extend({
+  image: '',
+  username: '',
+  bio: '',
+  email: '',
+  password: '',
+
+  actions: {
+    submit() {
+      const { image, username, bio, email, password } = this;
```
Comment (Alonski) [inline app/components/settings-form.js]: Hmm, should we be using `this.getProperties` instead of just `this` here?

**Comment by Alonski** [comment]
Thanks a lot @GCheung55

File: app/components/settings-form.js (hunk 1 above)
Comment (GCheung55) [reply app/components/settings-form.js]: I would switch to using `getProperties` if there were side effects to this approach.

**Pull request merged**